platform/o.n.swing.laf.flatlaf/src/org/netbeans/swing/laf/flatlaf/ui/FlatTabControlIcon.java

```diff
@@ -130,6 +131,21 @@ protected void paintIcon(Component c, Graphics2D g, int width, int height, doubl
         Color bgColor = new Color(0, 0, 0, 0); // Alpha zero means no background.
         Color fgColor = foreground;
         {
+            int buttonState = this.buttonState;
+            if (buttonState == -1) {
+                // get button state from component
+                if (!c.isEnabled()) {
+                    buttonState = TabControlButton.STATE_DISABLED;
```

Comment:
Have you tested that this state ends up being used? Are you sure the button does not end up trying to make its own disabled version using an ImageFilter? (In the latter case the disabled version would not look good on HiDPI screens.)

Reply:
Yes, this works. The reason is that class `FlatTabControlIcon` is not an instance of `javax.swing.ImageIcon`. So `LookAndFeel.getDisabledIcon()` returns `null` and the normal icon is used.

Reply:
Great!